File: app/src/main/java/im/dacer/kata/core/action/DuckDuckGoSearchAction.java
/*
 * The MIT License (MIT)
 * Copyright (c) 2016 baoyongzhang <baoyz94@gmail.com>
 */
package im.dacer.kata.core.action;

import android.net.Uri;

/**
 * Created by baoyongzhang on 2016/10/26.
 */
public class DuckDuckGoSearchAction extends SearchAction {

    public static DuckDuckGoSearchAction create() {
        return new DuckDuckGoSearchAction();
    }

    @Override
    public Uri createSearchUriWithEncodedText(String encodedText) {
        return Uri.parse("https://duckduckgo.com/?q=" + encodedText);
    }
}

File: app/src/main/java/im/dacer/kata/util/action/DuckDuckGoSearchAction.kt
package im.dacer.kata.util.action

import android.net.Uri

class DuckDuckGoSearchAction : WebSearchAction() {

    override fun createSearchUriWithEncodedText(encodedText: String?): Uri {
        return Uri.parse("https://duckduckgo.com/?q=" + encodedText!!)
    }

    companion object {
        fun create(): DuckDuckGoSearchAction {
            return DuckDuckGoSearchAction()
        }
    }
}
